[Javier Ong Ding Quan] IP

jeyvia opened the pull request:
Levels-1,2,3 completed so far, Level-4 wip.

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

LokQiJun commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+public class TaskList {
+    Task[] list = new Task[100];
```
Naming needs to be more specific

LokQiJun commented on src/main/java/Duke.java:
```diff
@@ -1,39 +1,25 @@
+            } else if (input.contains("unmark")) {
+                list.unmarkStatus(input);
+            } else if (input.contains("mark")) {
+                list.markStatus(input);
```
Consider using constants rather than magic strings

LokQiJun commented on src/main/java/Duke.java:
```diff
@@ -1,39 +1,25 @@
 import java.util.Scanner;
```
Good that you only imported the library that you need!

LokQiJun commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        System.out.println("Bye. Hope to see you again soon!");
+        printLine();
+    }
+    public void addTask(String input) {
```
Good that you followed the naming convention :)

LokQiJun reviewed (commented):
Very nicely organised :)

jingwei55 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+    }
+    public void unmarkStatus(String input) {
+        String[] instructions = input.split(" ");
+        int number = Integer.parseInt(instructions[1]);
+        list[number-1].isDone = false;
```
Can be more specific, perhaps could use index instead of number

jingwei55 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        printLine();
+    }
+
+    public void markStatus(String input) {
+        String[] instructions = input.split(" ");
+        int number = Integer.parseInt(instructions[1]);
+        list[number-1].isDone = true;
+        printLine();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  [X] " + list[number-1].description);
+        printLine();
+    }
```
Could be better if markStatus and unmarkStatus came from a more generic function. Both functions are very similar, hence there should be less repeat code

jingwei55 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,13 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Try to minimize the usage lambda functions if possible as they make code less readable

jingwei55 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        printLine();
+    }
+    public void addTask(String input) {
+        Task t = new Task(input);
+        list[index] = t;
```
Declaration of task can be more specific

jingwei55 reviewed (commented):
Overall the code is understandable, keep it up! Some issues need to be resolved to make your code more readable.

Commit:
Level-4

Commit:
Updating tags

jeyvia committed:
testing

jeyvia committed:
Merge branch 'branch-Level-5'

jeyvia committed:
9 sep commit

Not complete with exceptions

jeyvia committed:
Merge branch 'branch-Level-5'

jeyvia committed:
fixes

jeyvia committed:
Delete Task.java

jeyvia committed:
Level-5

jeyvia committed:
Level-6

jeyvia committed:
Level-7

okkhoy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+import consoleCommands.ConsoleCommands;
+import exception.*;
```
Please import classes explicitly

okkhoy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+                    System.out.println(TASK_DOES_NOT_EXIST_ERROR_MESSAGE);
+                    ConsoleCommands.printLine();
+                }
+            } else {
+                try {
+                    ConsoleCommands.addTask(input, taskList);
+                } catch (InvalidArgumentsException e) {
+                    System.out.println(INVALID_ARGUMENTS_ERROR_MESSAGE);
+                    ConsoleCommands.printLine();
+                } catch (InvalidCommandException e) {
+                    System.out.println(COMMAND_DOES_NOT_EXIST_ERROR_MESSAGE);
+                    ConsoleCommands.printLine();
+                } catch (NotEnoughArgumentsException e) {
+                    System.out.println(NOT_ENOUGH_ARGUMENTS_ERROR_MESSAGE);
+                    ConsoleCommands.printLine();
+                }
```
Oh! no no... this is not very good to default the behavior for a single command.

jeyvia replied:
Hi Prof! I'm not quite sure what it means to "default the behavior for a single command", could you kindly clarify? Thank you!

okkhoy commented on src/main/java/consoleCommands/ConsoleCommands.java:
```diff
@@ -0,0 +1,122 @@
+    public static final String MARKED_MESSAGE = "Nice! I've marked this task as done:";
+    public static final String UNMARKED_MESSAGE = "OK, I've marked this task as not done yet:";
+    public static final String HELLO_MESSAGE = "Hello! I'm Duke\n" + "What can I do for you?";
+    public static final String BYE_MESSAGE = "Bye. Hope to see you again soon!";
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+    public static void markStatus(String input, ArrayList<Task> Array) throws TaskDoesNotExistException {
+        printLine();
+        String[] instructions = input.split(" ");
```
Please improve spacing. Use the IDE to do it for you!

okkhoy commented on src/main/java/consoleCommands/ConsoleCommands.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("  " + Array.get(index-1));
+        printLine();
+    }
+    public static void printList(ArrayList<Task> Array) {
```
Parameter naming is off convention/standards.
If it is a constant it should be `ARRAY`, if not it should be `array`.
Also, a better name would be useful.

okkhoy commented on src/main/java/consoleCommands/ConsoleCommands.java:
```diff
@@ -0,0 +1,122 @@
+        System.out.println("Now you have " + Array.size() + " tasks in the list.");
+        printLine();
+    }
+    public static void addTask(String input, ArrayList<Task> Array) throws InvalidCommandException, InvalidArgumentsException, NotEnoughArgumentsException {
```
this line seems particularly long. try to wrap it

okkhoy commented on src/main/java/consoleCommands/ConsoleCommands.java:
```diff
@@ -0,0 +1,122 @@
+                throw new NotEnoughArgumentsException();
+            }
+            Array.add(new Event(eventInstructions[0],eventInstructions[1]));
+        }
+        else if (instructions[0].equals(COMMAND_TODO)) {
```
coding standard violation here.

okkhoy commented on src/main/java/consoleCommands/FileCommands.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.Scanner;
+
+public class FileCommands {
+    public static final String TRUE = "1";
+    public static final String FALSE = "0";
```
These variable names could trip the reader/maintainer, given that there is already keywords of the same name

okkhoy commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+package task;
+
+public class Deadline extends Task {
+    protected String by;
```
Why is this protected?
Are you inheriting anything else from deadline to make use of this field?

okkhoy commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,28 @@
+public class Task {
+    protected String description;
+    public boolean isDone;
+    public Task(String description) { //creating new object
```
Errrm... that is what a constructor is for.
Please remove that redundant comment.

okkhoy reviewed (commented):
Generally, the code is quite ok.
There are some specific things I have mentioned in the comments that help to improve readability/maintainability of the code. Please check out.

jeyvia committed:
iP edits

jeyvia committed:
Level-9

jeyvia committed:
A-MoreOOP

jeyvia committed:
Level-8

jeyvia committed:
A-JavaDoc

jeyvia committed:
A-UserGuide

jeyvia committed:
Update Duke.java

jeyvia committed:
Merge pull request #1 from jeyvia/branch-Level-8

Update Duke.java

jeyvia committed:
Final changes to iP

jeyvia committed:
Update Storage.java

jeyvia committed:
Added equalsIgnoreCase() for commands

jeyvia committed:
Checkstyle edits & comments

Edits done to code style after checkstyle
Added comments for storage > readFromFile
Edited UG

jeyvia committed:
Bug fixes, final changes and UG update

- Fixed bugs with date in event and deadline
- Fixed bugs with find command
- Fixed bugs with list command
- Minor changes to naming conventions and documentation
- Updated UG to fit new changes
- Added new exception DateParseException

jeyvia committed:
Update DateParseException.java
